src/client/index.ts

```diff
@@ -344,8 +367,16 @@ export async function create(options: NotificationOptions): Promise<Notification
         throw new Error('Invalid arguments passed to create: "date" must be a valid Date object');
     }
 
-    const response = await tryServiceDispatch(APITopic.CREATE_NOTIFICATION, {...options, date: options.date && options.date.valueOf()});
-    return {...response, date: new Date(response.date)};
+    if (options.expiry !== undefined && options.expiry !== null && !(options.expiry instanceof Date)) {
+        throw new Error('Invalid arguments passed to create: "expiry" must be null or a valid Date object');
+    }
+
+    const response = await tryServiceDispatch(APITopic.CREATE_NOTIFICATION, {
+        ...options,
+        date: options.date && options.date.valueOf(),
+        expiry: options.expiry && options.expiry.valueOf()
```

Comment:
Can this not be?:
```ts
date: options.date && new Date(options.date.valueOf()),
expiry: options.expiry && new Date(options.expiry.valueOf())
...
return response;
```
Doesn't the shortcircuit when failing return the value of the check, in this case null?

Reply:
The first object literal is what going to the provider (so  we want those to be numbers). What comes back will be numbers, that we want to convert to `Date`s for the client app in the second object literal